v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/stack-traces/stack-trace-solidity-errors.ts

```diff
@@ -280,27 +278,38 @@ function getMessageFromLastStackTraceEntry(
 
     case StackTraceEntryType.CONTRACT_CALL_RUN_OUT_OF_GAS_ERROR:
       return "Transaction reverted: contract call run out of gas and made the transaction revert";
+
+    /* These types are not expected to be the last entry in the stack trace, as
+    their presence indicates that another frame should follow in the call stack. */
+    case StackTraceEntryType.CALLSTACK_ENTRY:
+    case StackTraceEntryType.UNRECOGNIZED_CREATE_CALLSTACK_ENTRY:
+    case StackTraceEntryType.UNRECOGNIZED_CONTRACT_CALLSTACK_ENTRY:
+    case StackTraceEntryType.INTERNAL_FUNCTION_CALLSTACK_ENTRY:
+      return undefined;
   }
 }
 
-// TODO: see TODO at line 301
-// const inspect = Symbol.for("nodejs.util.inspect.custom");
-
-// Note: This error class MUST NOT extend ProviderError, as libraries
-//   use the code property to detect if they are dealing with a JSON-RPC error,
-//   and take control of errors.
+/**
+ * Note: This error class MUST NOT extend ProviderError, as libraries use the
+ * code property to detect if they are dealing with a JSON-RPC error, and take
+ * control of errors.
+ **/
 export class SolidityError extends Error {
-  public readonly stackTrace: SolidityStackTrace;
-
-  constructor(message: string, stackTrace: SolidityStackTrace) {
+  constructor(
+    message: string,
+    public readonly stackTrace: SolidityStackTrace,
+    public readonly data?: string,
+    public readonly transactionHash?: string,
+  ) {
     super(message);
-    this.stackTrace = stackTrace;
-  }
 
-  // TODO: Can we bring this back with isolated declarations?
-  // public [inspect](): string {
-  //   return this.inspect();
-  // }
+    Object.defineProperty(this, Symbol.for("nodejs.util.inspect.custom"), {
```

Comment:
Nice trick!

No need to define "inspect" in 314. We can use `value: () => { ... }`